run.sh

```diff
@@ -14,7 +14,26 @@ elif [ -n "${INIT_RESTORE_LATEST}" ]; then
   find /backup -maxdepth 1 -name '*.sql.gz' | tail -1 | xargs /restore.sh
 fi
 
+function final_backup {
+    echo "=> Captured trap for final backup"
+    DATE=$(date +%Y%m%d%H%M)
+    echo "=> Requested last backup at $(date "+%Y-%m-%d %H:%M:%S")"
+    exec /backup.sh
+    exit 0
+}
+
+if [ -n "${EXIT_BACKUP}" ]; then
+  echo "=> Listening on container shutdown gracefully to make last backup before close"
+  trap final_backup SIGHUP SIGINT SIGTERM
+fi
+
 echo "${CRON_TIME} /backup.sh >> /mysql_backup.log 2>&1" > /tmp/crontab.conf
 crontab /tmp/crontab.conf
 echo "=> Running cron task manager in foreground"
-exec crond -f -l 8 -L /mysql_backup.log
+crond -f -l 8 -L /mysql_backup.log &
+
+echo "Listening on crond, and wait..."
+
+while : ; do sleep 1 ; done
```

Comment:
Can we increase the number of seconds waiting in the loop? 1 seems to short to me

Reply:
It is the number of seconds the system waits to handle the shutdown signal (based on specific sleep implementation), so I think it is not a good idea to increase it. We could try with 2, but I think it will be less responsive.

Reply:
ok, but did you notice an increase in CPU consumption?

Reply:
Yes thanks, I checked and found a CPU consumption. I resolved using the following on line 37. It is a passive wait, so it is very better than the previous active wait. 
```
tail -f /dev/null & wait $!
```
In my test environment CPU goes from 0.14% to 0.01% (same as master image), I already committed that!